apps/website-25/src/components/courses/UnitFeedback.tsx

```diff
@@ -0,0 +1,147 @@
+      stroke="#FFAE36"
+      strokeWidth="2"
+      strokeLinecap="round"
+      strokeLinejoin="round"
+      fill={filled ? '#FFC16A' : 'transparent'}
+    />
+  </svg>
+);
+
+type UnitFeedbackProps = {
+  unit: Pick<Unit, 'id' | 'courseSlug'>;
+};
+
+const FREE_RESPONSE_PLACEHOLDER = '• What would have made this unit more valuable to you?\n• What aspects did you find most challenging?\n• What aspects did you find particularly useful?';
+
+const UnitFeedback: React.FC<UnitFeedbackProps> = ({ unit }) => {
+  const auth = useAuthStore((s) => s.auth);
+
+  const [rating, setRating] = useState<number>(0);
+  const [hoverRating, setHoverRating] = useState<number>(0);
+  const [feedbackText, setFeedbackText] = useState<string>('');
+  const [error, setError] = useState<string | null>(null);
+
+  const { courseSlug, id: unitId } = unit;
+
+  const [{ data }, refetch] = useAxios<GetUnitFeedbackResponse>({
+    method: 'get',
+    url: `/api/courses/${courseSlug}/${unitId}/feedback`,
+    headers: {
+      Authorization: `Bearer ${auth?.token}`,
+    },
+  }, { manual: !auth });
+
+  const handleSubmit = useCallback(async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!rating) {
+      setError('Please select a star rating');
+      return;
+    }
+
+    setError(null);
+
+    try {
+      await axios.put<unknown, unknown, PutUnitFeedbackRequest>(
+        `/api/courses/${courseSlug}/${unitId}/feedback`,
+        {
+          overallRating: rating,
+          anythingElse: feedbackText,
+        },
+        {
+          headers: {
+            Authorization: `Bearer ${auth?.token}`,
+          },
+        },
+      );
+
+      await refetch();
+    } catch (e) {
+      setError('Unexpected error');
+    }
+  }, [rating, feedbackText, courseSlug, unitId, auth, refetch]);
+
+  if (!data || !auth) {
+    return null;
+  }
+
+  const hasSubmitted = !!data.unitFeedback?.overallRating;
+
+  return (
+    <form
+      className="unit-feedback container-lined bg-white p-8 flex flex-col gap-6 mt-6"
+      onSubmit={handleSubmit}
+    >
+      <div className="unit-feedback__header flex flex-col gap-4">
+        <h4>How did you like this unit?</h4>
+        <p>
+          We constantly try to get better and need your feedback to improve the
+          course. It only takes <u>1 min</u> to share your learning experience
+          with us.
+        </p>
+      </div>
+      {hasSubmitted ? (
+        <p className="unit-feedback__sent-message flex gap-2 text-bluedot-dark items-center font-[650]">
```

Comment:
typography styles like text color should be set from the tailwind base layer by default
```suggestion
        <p className="unit-feedback__sent-message flex gap-2 items-center font-bold">
```